quartz/components/Explorer.tsx

```diff
@@ -35,11 +36,20 @@ export default ((userOpts?: Partial<Options>) => {
   let fileTree: FileNode
   let jsonTree: string
 
-  function constructFileTree(allFiles: QuartzPluginData[]) {
+  function constructFileTree(cfg: GlobalConfiguration, allFiles: QuartzPluginData[]) {
     if (!fileTree) {
       // Construct tree from allFiles
       fileTree = new FileNode("")
-      allFiles.forEach((file) => fileTree.add(file, 1))
+      for (const file of allFiles) {
```

Comment:
What about this ?
```js
// All in the forEach
allFiles.forEach((file) => {
    const slug = file.slug!

    let ignore = false
    for (const ignorePattern of cfg.unlistedPatterns) {
      if (slug.startsWith(ignorePattern)) { ignore = true; break } // break improve performances by not checking anymore if ignore is triggered once
    }
    if (!ignore) fileTree.add(file, 1)
})

// Compact (i did not tested performances, the first one might be more optimized)
allFiles.forEach((file) => {
    let ignore = cfg.unlistedPatterns.map( ignorePattern => (file.slug!).startsWith(ignorePattern) ).includes(true)
    if (!ignore) fileTree.add(file, 1)
})

// Really compact because why not ?
allFiles.forEach((file) => {
    if( !cfg.unlistedPatterns.map( ignorePattern => (file.slug!).startsWith(ignorePattern) ).includes(true) ) fileTree.add(file, 1) // else ignoring
})
```